frontend/src/pages/projects/screens/detail/storage/ConnectWorkspaceOptionsField.tsx

```diff
@@ -1,22 +1,42 @@
 import * as React from 'react';
-import { FormGroup, Radio, Select, Stack, StackItem } from '@patternfly/react-core';
+import { FormGroup, Radio, Select, SelectOption, Stack, StackItem } from '@patternfly/react-core';
 import { NotebookKind } from '../../../../../k8sTypes';
+import { getNotebooks } from '../../../../../api';
+import { ProjectDetailsContext } from '../../../ProjectDetailsContext';
+import { getNotebookDisplayName } from '../../../utils';
 
 type ConnectWorkspaceOptionsFieldProps = {
   fieldId: string;
-  allWorkspaces: NotebookKind[];
   selections: string[];
   setSelections: (selections: string[]) => void;
 };
 
 const ConnectWorkspaceOptionsField: React.FC<ConnectWorkspaceOptionsFieldProps> = ({
   fieldId,
-  allWorkspaces,
   selections,
   setSelections,
 }) => {
   const [workspaceSelectOpen, setWorkspaceSelectOpen] = React.useState<boolean>(false);
-  const [isConnectToAll, setConnectToAll] = React.useState<boolean>(true);
+  const [isConnectToAll, setConnectToAll] = React.useState<boolean>(false);
+  const { currentProject } = React.useContext(ProjectDetailsContext);
+  const [allWorkspaces, setAllWorkspaces] = React.useState<NotebookKind[]>([]);
+
+  React.useEffect(() => {
+    getNotebooks(currentProject.metadata.name)
+      .then((notebooks) => setAllWorkspaces(notebooks))
+      .catch((e) => console.error('Error getting notebooks: ', e));
+  }, [currentProject.metadata.name]);
+
+  const options = React.useMemo(
```

Comment:
You don't need to memo JSX creation... React already does this in the reconciler